src/renderer/pkijs/generatePems/generateRootCA.ts

```diff
@@ -0,0 +1,72 @@
+import { Integer, PrintableString, BitString } from 'asn1js'
+import { Certificate, AttributeTypeAndValue, BasicConstraints, Extension, Time, getCrypto } from 'pkijs'
+
+import { signAlg, hashAlg } from './config'
+import { generateKeyPair, dumpPEM } from './common'
+
+export const createRootCA = async (communityName) => {
+  const rootCA = await generateRootCA({
+    commonName: `${communityName} CA`,
+    signAlg,
+    hashAlg
+  })
+  await dumpCertificate(rootCA)
+}
+
+async function generateRootCA({ commonName, signAlg, hashAlg }) {
+  const basicConstr = new BasicConstraints({ cA: true, pathLenConstraint: 3 })
+  const keyUsage = getCAKeyUsage()
+  const certificate = new Certificate({
+    serialNumber: new Integer({ value: 1 }),
+    extensions: [
+      new Extension({
+        extnID: '2.5.29.19',
+        critical: false,
+        extnValue: basicConstr.toSchema().toBER(false),
+        parsedValue: basicConstr // Parsed value for well-known extensions
+      }),
+      new Extension({
+        extnID: '2.5.29.15',
+        critical: false,
+        extnValue: keyUsage.toBER(false),
+        parsedValue: keyUsage // Parsed value for well-known extensions
+      })
+    ],
+    notBefore: new Time({ type: 1, value: new Date() }),
+    notAfter: new Time({ type: 1, value: new Date(2022, 1, 1) })
```

Comment:
the date should not be hardcode;

we should either not have expiration date at all or calculate it relatively to current date

Reply:
This is still hardcoded. Have you seen this comment?